csvcubed/tests/unit/cli/inspect/test_inspectsparqlmanager.py

```diff
@@ -198,3 +225,65 @@ def test_select_dsd_code_list_and_cols_without_codelist_labels():
     assert len(result.codelists) == 3
     assert result.codelists[0].codeListLabel == ""
     assert result.codelists[0].colsInUsed == "Alcohol Sub Type"
+
+
+def test_select_qb_dataset_url():
+    """
+    Should return expected `DatasetURLResult`.
+    """
+    csvw_metadata_json_path = _test_case_base_dir / "datacube.csv-metadata.json"
+    metadata_processor = MetadataProcessor(csvw_metadata_json_path)
+    csvw_metadata_rdf_graph = metadata_processor.load_json_ld_to_rdflib_graph()
+
+    path = ROOT_DIR_PATH / "tests" / "test-cases" / "cli" / "inspect"
+
+    result: DatasetURLResult = select_qb_dataset_url(
+        csvw_metadata_rdf_graph,
+        f"file://{str(path)}/alcohol-bulletin.csv#dataset",
+    )
+    assert result.dataset_url == "alcohol-bulletin.csv"
+
+def test_select_single_unit_from_dsd():
+    """
+    Should return expected `DSDSingleUnitResult`.
+    """
+    csvw_metadata_json_path = _test_case_base_dir / "datacube.csv-metadata.json"
+    metadata_processor = MetadataProcessor(csvw_metadata_json_path)
+    csvw_metadata_rdf_graph = metadata_processor.load_json_ld_to_rdflib_graph()
+    dataset_uri = select_csvw_catalog_metadata(csvw_metadata_rdf_graph).dataset_uri
+
+    result: DSDSingleUnitResult = select_single_unit_from_dsd(
+        csvw_metadata_rdf_graph, dataset_uri, csvw_metadata_json_path
+    )
+    assert result.unit_label is None
+    assert result.unit_uri == "http://gss-data.org.uk/def/concept/measurement-units/{+unit}"
```

Comment:
This template suggests that there are multiple units in this dataset - the template implies the existence of a `unit` column containing different units.

Reply:
I was using the wrong input. I have refactored this test into the following two tests:

```
def test_select_single_unit_from_dsd_when_unit_label_is_not_none():
    """
    Should return expected `DSDSingleUnitResult`.
    """
    csvw_metadata_json_path = (
        _test_case_base_dir
        / "single-unit_multi-measure"
        / "final-uk-greenhouse-gas-emissions-national-statistics-1990-to-2020.csv-metadata.json"
    )
    metadata_processor = MetadataProcessor(csvw_metadata_json_path)
    csvw_metadata_rdf_graph = metadata_processor.load_json_ld_to_rdflib_graph()
    dataset_uri = select_csvw_catalog_metadata(csvw_metadata_rdf_graph).dataset_uri

    result: DSDSingleUnitResult = select_single_unit_from_dsd(
        csvw_metadata_rdf_graph, dataset_uri, csvw_metadata_json_path
    )
    assert result.unit_label == "MtCO2e"
    assert (
        result.unit_uri
        == "final-uk-greenhouse-gas-emissions-national-statistics-1990-to-2020.csv#unit/mtco2e"
    )

def test_select_single_unit_from_dsd_when_unit_label_is_none():
    """
    Should return expected `DSDSingleUnitResult`.
    """
    csvw_metadata_json_path = (
        _test_case_base_dir
        / "single-unit_single-measure"
        / "energy-trends-uk-total-energy.csv-metadata.json"
    )
    metadata_processor = MetadataProcessor(csvw_metadata_json_path)
    csvw_metadata_rdf_graph = metadata_processor.load_json_ld_to_rdflib_graph()
    dataset_uri = select_csvw_catalog_metadata(csvw_metadata_rdf_graph).dataset_uri

    result: DSDSingleUnitResult = select_single_unit_from_dsd(
        csvw_metadata_rdf_graph, dataset_uri, csvw_metadata_json_path
    )
    assert result.unit_label is None
    assert (
        result.unit_uri
        == "energy-trends-uk-total-energy.csv#unit/{+unit}"
    )
```